src/library_pthread.js

```diff
@@ -660,7 +660,11 @@ var LibraryPThread = {
   // allocations from __pthread_create_js we could also remove this.
   __pthread_create_js__noleakcheck: true,
   __pthread_create_js__sig: 'iiiii',
-  __pthread_create_js__deps: ['$spawnThread', 'pthread_self', '$pthreadCreateProxied'],
+  __pthread_create_js__deps: ['$spawnThread', 'pthread_self', '$pthreadCreateProxied',
+#if OFFSCREENCANVAS_SUPPORT
+    'malloc',
```

Comment:
Doesn't this need to be in deps_info? That is, if JS is the only thing that needs malloc, we need the deps_info mechanism, don't we?

Reply:
I did add this to deps info.  See the new entry for `pthread_create` below.   Not that its the use of `pthread_create` in users code that needs to go in deps_info not `__pthread_create_js` (which is called in directly from `pthread_create`.  Entries in deps_info always have to be symbols that exist in user object files.

Reply:
Also, note that I'm deliberately adding the dependency both here and in deps_info.  This one acts as a fail safe for cases where the deps_info for some reason didn't work.   If we don't add this one here then we get a runtime error instead of a build time error when deps_info isn't correct.